dvc/repo/reproduce.py

```diff
@@ -13,11 +13,19 @@
 def _reproduce_stage(stage, **kwargs):
     def _run_callback(repro_callback):
         _dump_stage(stage)
+        logger.debug(f"{repro_callback} ([{stage}])")
         repro_callback([stage])
 
     checkpoint_func = kwargs.pop("checkpoint_func", None)
-    if checkpoint_func:
-        kwargs["checkpoint_func"] = partial(_run_callback, checkpoint_func)
+    if stage.is_checkpoint:
+        if checkpoint_func:
+            kwargs["checkpoint_func"] = partial(_run_callback, checkpoint_func)
+        else:
+            logger.warning(
+                "Checkpoint stages are not fully supported in 'dvc repro'. "
+                "Checkpoint stages should be reproduced with 'dvc exp run' "
+                "or 'dvc exp resume'."
+            )
```

Comment:
I considered erroring out here and went with the warning in the end, but I don't feel too strongly either way.

Running a stage w/checkpoint outs from `dvc repro` won't actually break anything, it will just be handled the same as a stage with persist outs - the main difference being that depending on the user code this may be a command that runs forever until manually killed.